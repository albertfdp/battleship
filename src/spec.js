const expect = require('test/expect');
const { createBoard } = require('.');

describe('createBoard', () => {
  let board;

  beforeEach(() => {
    board = createBoard([5, 4, 4, 3, 2, 1]);
  });

  it('creates a board with 100 cells', () => {
    expect(board.cells.size, 'to equal', 100);
  });

  it('creates a board with 4 boats', () => {
    expect(board.boats.size, 'to equal', 6);
  });

  it('prints the board with no boats', () => {
    expect(board.print(), 'to match snapshot');
  });
});
