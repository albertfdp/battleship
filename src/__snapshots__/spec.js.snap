// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createBoard prints the board with no boats 1`] = `
"
----1----2----3----4----5----6----7----8----9----10-
A |    |    |    |    |    |    |    |    |    |   |
----------------------------------------------------
B |    |    |    |    |    |    |    |    |    |   |
----------------------------------------------------
C |    |    |    |    |    |    |    |    |    |   |
----------------------------------------------------
D |    |    |    |    |    |    |    |    |    |   |
----------------------------------------------------
E |    |    |    |    |    |    |    |    |    |   |
----------------------------------------------------
F |    |    |    |    |    |    |    |    |    |   |
----------------------------------------------------
G |    |    |    |    |    |    |    |    |    |   |
----------------------------------------------------
H |    |    |    |    |    |    |    |    |    |   |
----------------------------------------------------
I |    |    |    |    |    |    |    |    |    |   |
----------------------------------------------------
J |    |    |    |    |    |    |    |    |    |   |
----------------------------------------------------"
`;
